packages/react-reconciler/src/__tests__/ReactSuspenseFallback-test.js

```diff
@@ -0,0 +1,212 @@
+ *
+ * @flow
+ */
+let React;
+let ReactTestRenderer;
+let ReactFeatureFlags;
+let Scheduler;
+let ReactCache;
+let Suspense;
+
+let TextResource;
+let textResourceShouldFail;
+
+describe('ReactSuspenseFallback', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    ReactFeatureFlags = require('shared/ReactFeatureFlags');
+
+    ReactFeatureFlags.replayFailedUnitOfWorkWithInvokeGuardedCallback = false;
+    React = require('react');
+    ReactTestRenderer = require('react-test-renderer');
+    Scheduler = require('scheduler');
+    ReactCache = require('react-cache');
+
+    Suspense = React.Suspense;
+
+    TextResource = ReactCache.unstable_createResource(
+      ([text, ms = 0]) => {
+        let listeners = null;
+        let status = 'pending';
+        let value = null;
+        return {
+          then(resolve, reject) {
+            switch (status) {
+              case 'pending': {
+                if (listeners === null) {
+                  listeners = [{resolve, reject}];
+                  setTimeout(() => {
+                    if (textResourceShouldFail) {
+                      Scheduler.unstable_yieldValue(
+                        `Promise rejected [${text}]`,
+                      );
+                      status = 'rejected';
+                      value = new Error('Failed to load: ' + text);
+                      listeners.forEach(listener => listener.reject(value));
+                    } else {
+                      Scheduler.unstable_yieldValue(
+                        `Promise resolved [${text}]`,
+                      );
+                      status = 'resolved';
+                      value = text;
+                      listeners.forEach(listener => listener.resolve(value));
+                    }
+                  }, ms);
+                } else {
+                  listeners.push({resolve, reject});
+                }
+                break;
+              }
+              case 'resolved': {
+                resolve(value);
+                break;
+              }
+              case 'rejected': {
+                reject(value);
+                break;
+              }
+            }
+          },
+        };
+      },
+      ([text, ms]) => text,
+    );
+    textResourceShouldFail = false;
+  });
+
+  function Text(props) {
+    Scheduler.unstable_yieldValue(props.text);
+    return props.text;
+  }
+
+  function AsyncText(props) {
+    const text = props.text;
+    try {
+      TextResource.read([props.text, props.ms]);
+      Scheduler.unstable_yieldValue(text);
+      return text;
+    } catch (promise) {
+      if (typeof promise.then === 'function') {
+        Scheduler.unstable_yieldValue(`Suspend! [${text}]`);
+      } else {
+        Scheduler.unstable_yieldValue(`Error! [${text}]`);
+      }
+      throw promise;
+    }
+  }
+
+  it('suspends and shows fallback', () => {
+    // Render an empty shell
+    const root = ReactTestRenderer.create(
```

Comment:
Use noop renderer please so that we get coverage of Fabric. It's the only renderer that gets tested in both mutation and persistent mode.

Reply:
Ah, I see, I was going off [this comment](https://github.com/facebook/react/blob/48c9c77ecfbd275903cfda8af851db6ffe01250f/packages/react-reconciler/src/__tests__/ReactSuspense-test.internal.js#L12-L13).

Reply:
Let's remove any such files if you see them. That was an outdated strategy.

Reply:
Ah yeah can you delete that while you're at it? :D

Long term we want to replace the React Noop renderer with a unified testing API that can be used with all our renderers. Then all tests would run in all renderers.

Reply:
Got it thanks for the context! Updated and removed those comments.